firebase-firestore/src/main/java/com/google/firebase/firestore/model/Values.java

```diff
@@ -264,8 +265,11 @@ private static int compareTimestamps(Timestamp left, Timestamp right) {
   }
 
   private static int compareReferences(String leftPath, String rightPath) {
-    List<String> leftSegments = Splitter.on('/').splitToList(leftPath);
-    List<String> rightSegments = Splitter.on('/').splitToList(rightPath);
+    checkNotNull(leftPath);
+    checkNotNull(rightPath);
+    List<String> leftSegments = Lists.newArrayList(leftPath.split("/"));
```

Comment:
Nit: This is a private method. I don't think we need to do null-checks here. `rightPath` and `leftPath` would only be null if the Proto is invalid, in which case `leftPath.split()` would throw the same NPE that you are checking for manually.

Reply:
For background: `checkNotNull` is only useful if you're retaining an argument in a method but not actually using it there. The best example is if you're taking a constructor argument and assigning it to an instance variable without using it. The idea is to force the NPE to happen at the point where the caller is apparent.